src/ProxyManager/ProxyGenerator/NullObject/MethodGenerator/StaticProxyConstructor.php

```diff
@@ -35,6 +35,7 @@ function (ReflectionProperty $publicProperty) : string {
             Properties::fromReflectionClass($originalClass)->getPublicProperties()
         );
 
+        $this->setReturnType($this->getName());
```

Comment:
Ran the tests locally: `$this->getName()` is the method name, not the class name :S